Cairo Desktop/Cairo Desktop/DesktopNavigationToolbar.xaml.cs

```diff
@@ -28,15 +33,56 @@ public Desktop ToolbarOwner
         public DesktopNavigationToolbar()
         {
             InitializeComponent();
+
             SetPosition();
 
             browseContextMenu = new System.Windows.Controls.ContextMenu();
+
+            lowLevelKeyboardListener = new Common.LowLevelKeyboardListener();
+            lowLevelKeyboardListener.HookKeyboard();
+            lowLevelKeyboardListener.OnKeyDown += LowLevelKeyboardListener_OnKeyDown;
+            lowLevelKeyboardListener.OnKeyUp += LowLevelKeyboardListener_OnKeyUp;
+        }
+
+        bool backButtonTypeToggled = false;
+        private void LowLevelKeyboardListener_OnKeyDown(object sender, Common.KeyEventArgs e)
+        {
+            if (IsMouseOver)
+            {
+                if (e.Key == System.Windows.Input.Key.LeftShift || e.Key == System.Windows.Input.Key.RightShift)
+                {
+                    backButtonImg.Source = new BitmapImage(new Uri("/Resources/controlsParentFolder.png", UriKind.RelativeOrAbsolute));
+                    backButtonTypeToggled = true;
+                    btnBack.ToolTip = "Parent";
+                }
+            }
+        }
+
+        private void LowLevelKeyboardListener_OnKeyUp(object sender, Common.KeyEventArgs e)
+        {
+            if (backButtonTypeToggled)
+            {
+                if (e.Key == System.Windows.Input.Key.LeftShift || e.Key == System.Windows.Input.Key.RightShift)
+                {
+                    backButtonTypeToggled = false;
+                    backButtonImg.Source = new BitmapImage(new Uri("/Resources/controlsBack.png", UriKind.RelativeOrAbsolute));
+                    btnBack.ToolTip = "Back";
+                }
+            }
         }
 
         private void SetPosition()
         {
-            Top = AppBarHelper.PrimaryMonitorSize.Height - Height - 150;
-            Left = (AppBarHelper.PrimaryMonitorSize.Width / 2) - (Width / 2);
+            if (Settings.Instance.DesktopNavigationToolbarLocation != default(System.Windows.Point))
+            {
+                Top = Settings.Instance.DesktopNavigationToolbarLocation.Y;
```

Comment:
Let's say I have a docked laptop. What happens if I save the toolbar position in the bottom right corner while docked on a large screen, quit Cairo, then launch Cairo on the small screen? Is the toolbar off screen or does it default to center?

Reply:
It would show off screen. I have fixed